components/project-list/project-list.jsx

```diff
@@ -16,28 +16,48 @@ export default React.createClass({
       displayBatchIndex: 1
     };
   },
+  getDefaultProps() {
+    return {
+      params: {
+        featured: null,
+        issue: null,
+        search: null,
+        ids: null
+      }
+    };
+  },
   componentDidMount() {
     this.fetchData(this.props.params);
   },
   componentWillReceiveProps(nextProps) {
+    // Reset state
+    this.resetState();
```

Comment:
But most of the time you won't even have a chance to read that text if the connection is reasonably fast- this process results in a "flicker" of an intermediate state, instead of just changing from one set of results to another. If you want something in the middle, might I suggest a spinner or something to indicate something's happening in the background, without immediately clearing the results?

Think of the case where someone searches for "web" and then decides to add "literacy" to the end of it- it doesn't make sense that the results disappear before reappearing- they should just reduce to entries with the complete string "web literacy" in them.